File: ListView/ListViewSample/app/src/main/java/com/example/nabha/listviewsample/MainActivity.java
package com.example.nabha.listviewsample;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

import static android.support.v7.appcompat.R.styleable.View;
import static com.example.nabha.listviewsample.R.id.parent;

public class MainActivity extends AppCompatActivity {

    private ListView mListView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mListView = (ListView) findViewById(R.id.recipe_list_view);
        final ArrayList<Recipe> recipeArrayList = Recipe.getRecipesFromFile("recipes.json", this);
        RecipeAdapter adapter = new RecipeAdapter(getApplicationContext(), recipeArrayList);
        mListView.setAdapter(adapter);

        final Context context = this;
        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {

            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                // 1
                Recipe selectedRecipe = recipeArrayList.get(position);

                // 2
                Intent detailIntent = new Intent(context, RecipeDetailActivity.class);

                // 3
                detailIntent.putExtra("title", selectedRecipe.title);
                detailIntent.putExtra("url", selectedRecipe.instructionUrl);

                // 4
                startActivity(detailIntent);
            }

        });
    }
}

File: KExtensions/app/src/main/java/com/example/kextensions/MainActivity.kt
package com.example.kextensions

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val occurence = "Abba Dabba Jabba".occurrence()

    }

    private var characterMap = HashMap<Char, Int>()


    private fun String.occurrence(): String {
        val str = this.removeWhiteSpace().toCharArray()
        var occurrenceString = ""

        for (c in str) {
            if (characterMap.containsKey(c)) {
                characterMap[c] = (characterMap[c] ?: 1) + 1
            } else {
                characterMap[c] = 1
            }
        }

        characterMap.forEach {
            occurrenceString = "$occurrenceString\n${it.key}: ${it.value}"
        }

        return occurrenceString
    }

    private fun String.removeWhiteSpace(): String {
        return this.replace(" ", "")
    }
}
